practice-two/src/components/FormValidation/FormValidation.jsx

```diff
@@ -0,0 +1,186 @@
+import { useCallback, useState } from 'react';
+import './FormValidation.css';
+import { InputValidate, Button } from '@components';
+import { BUTTON_VARIANTS } from '@constants';
+import { Validation } from '@helpers';
+
+const FormValidation = (props) => {
+  const { handleShowForm } = props;
+  const [formData, setFormData] = useState({
+    username: '',
+    avatar: '',
+    email: '',
+    phonenumber: '',
+    description: '',
+    address: '',
+    gender: 'male',
+  });
+
+  const [errors, setErrors] = useState({
+    username: '',
+    avatar: '',
+    email: '',
+    phonenumber: '',
+    description: '',
+    address: '',
+  });
+
+  const handleChange = useCallback((e) => {
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [e.target.name]: e.target.value,
+    }));
+  }, []);
+
+  const handleBlur = useCallback((e) => {
+    const { name, value } = e.target;
+    const errorMessage = Validation(name, value);
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      [name]: errorMessage,
+    }));
+  }, []);
+
+  const handleSubmit = useCallback(
+    (e) => {
+      e.preventDefault();
+
+      const newErrors = {};
+      for (const field in formData) {
+        const errorMessage = Validation(field, formData[field]);
+        if (errorMessage) {
+          newErrors[field] = errorMessage;
+        }
+      }
+      setErrors(newErrors);
+
+      if (Object.keys(newErrors).length === 0) {
+        console.log(formData);
+      }
+    },
+    [formData, setErrors]
+  );
+
+  return (
+    <div className='form__background'>
+      <form className='form__validation' onSubmit={handleSubmit}>
+        <p className='form__title'>Add customer</p>
+        <div className='form__row'>
+          <div className='col-6'>
+            <div className='form__input'>
+              <label className='input__label'>Full name</label>
```

Comment:
create component inclide label and input

Reply:
![image](https://github.com/datdragon777/React-Training/assets/74806826/87efd57e-74f5-409b-bc22-3c97c8b57e35)